modules/renter/siafile/metadata.go

```diff
@@ -0,0 +1,288 @@
+package siafile
+
+import (
+	"math"
+	"os"
+	"time"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+type (
+	// Metadata is the metadata of a SiaFile and is JSON encoded.
+	Metadata struct {
+		version      [16]byte          // version of the sia file format used
+		fileSize     int64             // total size of the file
+		masterKey    crypto.TwofishKey // masterkey used to encrypt pieces
+		pieceSize    uint64            // size of a single piece of the file
+		trackingPath string            // file to the local copy of the file used for repairing
+		siaPath      string
+
+		// The following fields are the usual unix timestamps of files.
+		modTime    time.Time // time of last content modification
+		changeTime time.Time // time of last metadata modification
+		accessTime time.Time // time of last access
+		createTime time.Time // time of file creation
+
+		// File ownership/permission fields.
+		mode os.FileMode // unix filemode of the sia file - uint32
+		uid  int         // id of the user who owns the file
+		gid  int         // id of the group that owns the file
+
+		// chunkHeaderSize is the size of each of the following chunk's metadata.
+		chunkHeaderSize uint64
+		// chunkBodySize is the size of each of the following chunk's bodies.
+		chunkBodySize uint64
+
+		// The following fields are the offsets for data that is written to disk
+		// after the pubKeyTable. We reserve a generous amount of space for the
+		// table and extra fields, but we need to remember those offsets in case we
+		// need to resize later on.
+		//
+		// chunkOffset is the offset of the first chunk, forced to be a factor of
+		// 4096, default 16kib
```

Comment:
are you sure that a default of 4096 is not good enough? A default of 16kib means that there's going to be a lot of metadata overhead for tiny files.

So much actually that it makes sense to have really small files (<64kb) just to be stored entirely instead of the chunks, which I think is probably the best solution here, heh.

Reply:
I think that is just a possible value we talked. We can also do 4kib instead.